cmd/cloud.go

```diff
@@ -303,24 +318,16 @@ This will execute the test on the k6 cloud service. Use "k6 login cloud" to auth
 				if progressErr == nil {
 					if (newTestProgress.RunStatus > lib.RunStatusRunning) ||
 						(exitOnRunning && newTestProgress.RunStatus == lib.RunStatusRunning) {
-						shouldExitLoop = true
+						globalCancel()
```

Comment:
Hmm this change doesn't seem right, I think you should keep `shouldExitLoop` here. 

Imagine that you have a slow connection or the k6 cloud backend is slow and `client.GetTestProgress(refID)` takes more than 2 seconds (the `ticker` period). Then, on the next iteration of the `for {}` loop, the `select` has an equal chance to pick either branch, and you can end up back here again, rather than breaking out of the loop

Reply:
I think this should just call `ticker.Stop()` ... There ~was~[is a bug](https://github.com/golang/go/issues/37196) where it might send one more tick (if it has alredy ticked basically) but I think we can ignore that or ... drain it with the `select/default` trick

Reply:
As you say @MStoykov, it can tick one more time if the `client.GetTestProgress(refID)` call took more than 2 seconds. I think cleanest solutions is both the previous logic there (`shouldExitLoop` and breaking the loop as soon as that `if` succeeds) **and** cancelling the context.

Reply:
Yes @theerapatcha , that is more or less what @na-- wrote, although I agree that it might've not been so clear :(

so you just need to add a `break` on the next line here and we should be ready to merge this :)

Reply:
done :)